src/tools/rbd/action/Export.cc

```diff
@@ -540,7 +553,7 @@ static int do_export_v1(librbd::Image& image, librbd::image_info_t &info,
   file_size += info.size;
   r = throttle.wait_for_ret();
   if (fd != 1) {
-    if (r >= 0) {
+    if (r >= 0 && !is_blk_dev) {
```

Comment:
... and if the block device passed to `rbd export` is smaller than the image, it seems like work would be started anyway only to fail in the middle?

What is the use case for exporting to a block device, anyway?  It doesn't seem well defined or very useful to me: if the size changes as a result of exporting the filesystem(s) on top might get confused for example.  Was it an actual ask from a user or is it being added just for symmetry with importing from a block device?

Reply:
Can these be dealt with by mapping an image with rbd-wnbd and using an external dd-like tool for a copy?  This way a user would be presented with mismatched sizes and other potential oddities and forced to make explicit choices.

Reply:
Sure, although it's not very user friendly, especially since there's no built-in ``dd`` equivalent on Windows.

Reply:
> Somewhat related -- have you considered the case where the block device passed to rbd export is larger than the image, possibly with some junk data at the end? If it's then passed to rbd import for a round trip, would the imported image also be larger (and contain that junk data)?

Indeed, in that case it would ignore the junk data at the end. The alternative would be to clear the remaining disk space but I'm not sure if it's worth going there.

Reply:
> Sure, although it's not very user friendly, especially since there's no built-in `dd` equivalent on Windows.

I'd say that `rbd export -> <some block device> -> rbd import` and having imported image be larger (something clearing the remaining disk space wouldn't help with) isn't particularly user friendly either.  How about requiring that the destination block device has the same size?

Reply:
Sounds reasonable.

Reply:
I've added a size check, ensuring that the image size matches the block device size.